ui/dashboards/src/components/Variables/Variable.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useEffect, useRef } from 'react';
+import { Select, FormControl, InputLabel, MenuItem, Box, LinearProgress, IconButton, TextField } from '@mui/material';
+import { Reload } from 'mdi-material-ui';
+import { VariableName, ListVariableDefinition } from '@perses-dev/core';
+import { useTemplateVariable, useTemplateVariableActions, useTemplateVariableSrv } from '../../context';
+
+type TemplateVariableProps = {
+  name: VariableName;
+};
+
+export function TemplateVariable({ name }: TemplateVariableProps) {
+  const ctx = useTemplateVariable(name);
+  const kind = ctx.definition?.kind;
+  switch (kind) {
+    case 'TextVariable':
+      return <TextVariable name={name} />;
+    case 'ListVariable':
+      return <ListVariable name={name} />;
+  }
+
+  return <div>Unsupported Variable Kind: ${kind}</div>;
+}
+
+function ListVariable({ name }: TemplateVariableProps) {
+  const ctx = useTemplateVariable(name);
+  const definition = ctx.definition as ListVariableDefinition;
+  const { setVariableValue, loadTemplateVariable } = useTemplateVariableActions();
+  const allowMultiple = definition?.options.allowMultiple === true;
+  useEffect(() => {
+    loadTemplateVariable(name);
+  }, [name, loadTemplateVariable]);
+
+  let value = ctx.state?.value;
+
+  if (allowMultiple && !Array.isArray(value)) {
+    value = [];
+  }
```

Comment:
it'd be cool if we could set this default somehow in types, but not sure that's possible

Reply:
Yeah good call, I think we can add a default. I'll probably do that in the next PR the implements List variable loading.